components/brave_wallet_ui/components/extension/transaction-detail-panel/index.tsx

```diff
@@ -0,0 +1,230 @@
+
+export interface Props {
+  transaction: BraveWallet.TransactionInfo
+  selectedNetwork: BraveWallet.EthereumChain
+  accounts: WalletAccountType[]
+  visibleTokens: BraveWallet.ERCToken[]
+  transactionSpotPrices: BraveWallet.AssetPrice[]
+  defaultCurrencies: DefaultCurrencies
+  onBack: () => void
+  onRetryTransaction: (transaction: BraveWallet.TransactionInfo) => void
+  onSpeedupTransaction: (transaction: BraveWallet.TransactionInfo) => void
+  onCancelTransaction: (transaction: BraveWallet.TransactionInfo) => void
+}
+
+const TransactionDetailPanel = (props: Props) => {
+  const {
+    transaction,
+    selectedNetwork,
+    accounts,
+    visibleTokens,
+    transactionSpotPrices,
+    defaultCurrencies,
+    onBack,
+    onRetryTransaction,
+    onSpeedupTransaction,
+    onCancelTransaction
+  } = props
+
+  const parseTransaction = useTransactionParser(selectedNetwork, accounts, transactionSpotPrices, visibleTokens)
+  const transactionDetails = React.useMemo(
+    () => parseTransaction(transaction),
+    [transaction]
+  )
+
+  const fromOrb = React.useMemo(() => {
+    return EthereumBlockies.create({ seed: transactionDetails.sender.toLowerCase(), size: 8, scale: 16 }).toDataURL()
+  }, [transactionDetails.sender])
+
+  const toOrb = React.useMemo(() => {
+    return EthereumBlockies.create({ seed: transactionDetails.recipient.toLowerCase(), size: 8, scale: 16 }).toDataURL()
+  }, [transactionDetails.recipient])
+
+  const onClickViewOnBlockExplorer = () => {
+    const explorerURL = selectedNetwork.blockExplorerUrls[0]
+    if (explorerURL && transaction?.txHash) {
+      const url = `${explorerURL}/tx/${transaction.txHash}`
+      window.open(url, '_blank')
+    } else {
+      alert(getLocale('braveWalletTransactionExplorerMissing'))
+    }
+  }
+
+  const onClickRetryTransaction = () => {
+    if (transaction) {
+      onRetryTransaction(transaction)
+    }
+  }
+
+  const onClickSpeedupTransaction = () => {
+    if (transaction) {
+      onSpeedupTransaction(transaction)
+    }
+  }
+
+  const onClickCancelTransaction = () => {
+    if (transaction) {
+      onCancelTransaction(transaction)
+    }
+  }
+
+  const transactionTitle = React.useMemo((): string => {
+    if (transactionDetails.isSwap) {
+      return toProperCase(getLocale('braveWalletSwap'))
+    }
+    if (transaction.txType === BraveWallet.TransactionType.ERC20Approve) {
+      return toProperCase(getLocale('braveWalletApprovalTransactionIntent'))
+    }
+    return toProperCase(getLocale('braveWalletTransactionSent'))
+  }, [transactionDetails, transaction])
+
+  const transactionValue = React.useMemo((): string => {
+    if (transaction.txType === BraveWallet.TransactionType.ERC721TransferFrom ||
+      transaction.txType === BraveWallet.TransactionType.ERC721SafeTransferFrom) {
+      return transactionDetails.erc721ERCToken?.name + ' ' + transactionDetails.erc721TokenId
+    }
+    if (transactionDetails.value === 'Unlimited') {
```

Comment:
This won't work on non-english locales because the string is localized. Maybe we should add a case to `formatWithCommasAndDecimals` like this:

```ts
if (isNaN(valueToNumber)) {
  return value
}
```

Reply:
Good catch! Updated `formatWithCommasAndDecimals` util and test.